std/conv.d

```diff
@@ -875,11 +875,19 @@ T toImpl(T, S)(S value)
         ()@trusted{ memcpy(result.ptr, value.ptr, value.length); }();
         return cast(T) result;
     }
-    else static if (isPointer!S && is(S : const(char)*))
+    else static if (isPointer!S && (is(S : const(char)*) || is(S : const(wchar)*) || is(S : const(dchar)*)))
     {
-        import core.stdc.string : strlen;
-        // It is unsafe because we cannot guarantee that the pointer is null terminated.
-        return value ? cast(T) value[0 .. strlen(value)].dup : null;
+        // This is unsafe because we cannot guarantee that the pointer is null terminated.
```

Comment:
That would require factoring out that part into a separate function. The other parts of the function are not @system, and due to templates they will not be marked @system if the instantiated part doesn't do anything shady. This is exactly what attribute inference is for.